src/ir/named.rs

```diff
@@ -257,7 +257,7 @@ pub struct UsedTemplateParameters<'ctx, 'gen>
     where 'gen: 'ctx
 {
     ctx: &'ctx BindgenContext<'gen>,
-    used: HashMap<ItemId, ItemSet>,
+    used: HashMap<ItemId, Option<ItemSet>>,
```

Comment:
Perhaps a comment here saying that the `Option` here is just for moving out of the hashmap and in again would be helpful.